### PR TITLE
Introduce Membership type

### PR DESCRIPTION
**Pull request by stas-demydiuk**
Replace strings with Membership type from [this PR](https://github.com/matrix-org/matrix-js-sdk/pull/2723)

Signed-off-by: Stanislav Demydiuk [s.demydiuk@gmail.com](mailto:s.demydiuk@gmail.com)

## Checklist

* [x] Tests written for new code (and old code if feasible) - Not applicable
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: cypress/e2e/room-directory/room-directory.spec.ts [hunk 1]
```diff
@@ -96,7 +98,7 @@ describe("Room Directory", () => {
         cy.get(".mx_RoomDirectory_dialogWrapper").contains(".mx_RoomDirectory_listItem", name)
             .should("exist").as("resultRow");
         cy.get(".mx_RoomDirectory_dialogWrapper").percySnapshotElement("Room Directory - filtered one result");
-        cy.get("@resultRow").find(".mx_AccessibleButton").contains("Join").click();
+        cy.get("@resultRow").find(".mx_AccessibleButton").contains(Membership.Join).click();
```
Comment (SimonBrandner) [inline cypress/e2e/room-directory/room-directory.spec.ts:101]: This is checking for a button label...

**Review by SimonBrandner — APPROVED** [review]
LGTM otherwise

File: cypress/e2e/room-directory/room-directory.spec.ts (hunk 1 above)
Comment (stas-demydiuk) [reply cypress/e2e/room-directory/room-directory.spec.ts]: My bad. Good catch, thank you!

Comment (SimonBrandner) [reply cypress/e2e/room-directory/room-directory.spec.ts]: No worries

**Comment by andybalaam** [comment]
Thanks again @stas-demydiuk for this PR and your persistence working on it. As mentioned in https://github.com/matrix-org/matrix-js-sdk/pull/2723 , I have taken the work you did and adapted it to use an enum unioned with a string.

My new PR based on this one is https://github.com/matrix-org/matrix-react-sdk/pull/12336

**Comment by andybalaam** [comment]
Closed in favour of https://github.com/matrix-org/matrix-react-sdk/pull/12336

**Pull request closed without merging**